### PR TITLE
feat: individual copy links in add context, general files box cleanup

### PR DESCRIPTION
**Pull request by jessicaschilling**
## In this PR

- Closes https://github.com/ipfs-shipyard/ipfs-share-files/issues/59 by adding individual "copy" links to files listed in the "add files" context

- Closes https://github.com/ipfs-shipyard/ipfs-share-files/issues/94 by switching to `dweb.link`in endpoints

Also the following cosmetic and IA improvements:
- In "add" context, strengthens visual weight of "Add files" section and improves consistency of padding/margins
- In "add" context, adds "Share individual files" and "Share all files" headers for better information hierarchy
- Rearranges titles, subtitles in "add" context for clarity
- Improves "this link changes" explanatory text
- Fixes file icon scaling flexbox issue when file name is long
- In "download" context, boosts prominence of "Download all" button

### In "add" context:
- [x] Remove repetition of “share individual files”
image

### In "download" context:
- [x] Remove individual "copy" buttons
- [x] Remove “share individual files” text
~~- [ ] Add big “Get files” header~~ <- decided against this as it'll be redundant once we make changes in https://github.com/ipfs-shipyard/ipfs-share-files/issues/85
image

### CODE REVIEW

**Comment by jessicaschilling** [comment]
@lidel @terichadbourne @rafaelramalho19 - this is still WIP but wanted to let y'all know my current status as of EOD Monday. Thanks!

**Comment by lidel** [comment]
Huge :+1:  for copying direct links to specific files

- I imagine the remaining work is to figure out if copy action for each file (red arrow) either needs different color or be an icon to differentiate it from the global Copy action (green arrow). Right now it is a bit unclear, there is no "default action": 
  > 

- suggestion: direct copied links should point at specific CID, so `http://dweb.link/ipfs/{cid}?filename={name}` – this way each copied direct link  is independent of the root folder and other files, making it more useful for ad-hoc sharing.

**Comment by jessicaschilling** [comment]
@lidel I *think* the most recent set of commits takes care of what you mention in https://github.com/ipfs-shipyard/ipfs-share-files/pull/89#issuecomment-771862732 - but can you please have a review? Thank you.

File: src/components/file/File.js [hunk 1]
```diff
@@ -99,30 +100,63 @@ export class File extends React.Component {
...
+  }
+
   render () {
     const { name, type, shareCID, error, t } = this.props
     const size = filesize(this.props.size || 0, { round: 0, spacer: '' })
 
     const fileNameClass = classnames({ charcoal: !error, gray: error }, ['FileLinkName ph2 f6 b truncate'])
     const fileSizeClass = classnames({ 'charcoal-muted': !error, gray: error }, ['f6'])
 
+    const url = `${ENDPOINTS.gateway}/${shareCID}?filename=${encodeURI(name)}`
```
Comment (lidel) [inline src/components/file/File.js:129]: When `?filename` is used we need to replace `shareCID` (CID of wrapper  directory) with the CID of the file.
I think its `cid` inside of `this.props`?

perhaps 

```diff
-      const { name, type, shareCID, error, t } = this.props 
+     const { name, type, cid, error, t } = this.props 
```

And then `cid.toString('base32')` (to stabilize CID so it looks the same on path and subdomain)

**Review by lidel — CHANGES_REQUESTED** [review]

File: src/components/file/File.js (hunk 1 above)
Comment (rafaelramalho19) [reply src/components/file/File.js:129]: correct, it's cid

**Review by rafaelramalho19 — APPROVED** [review]
LGTM, just need that change that Lidel pointed out :)

File: src/components/file/File.js (hunk 1 above)
Comment (jessicaschilling) [reply src/components/file/File.js]: This makes sense, but I'm afraid I can't find anywhere to `cid.toString('base32')` without getting "undefined" errors.

Comment (lidel) [reply src/components/file/File.js]: Seems that `cid` is not always present when render is executed –  this becames a problem if we want to calle `toV1` on it (to ensure its always something we can represent as base32), so I just added additional handling – should work fine now.

**Review by lidel — APPROVED** [review]
I've switched to dweb.link and CIDv1 while we are at it, so stable CIDs are copied.
Merge at will.

**Pull request merged**